src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -70,6 +77,55 @@ public function __construct(QueryBuilder $query)
     }
 
     /**
+     * Register a new global scope.
+     *
+     * @param  \Illuminate\Database\Eloquent\ScopeInterface|\Closure  $scope
+     * @param  string  $identifier
+     * @return $this
+     */
+    public function applyGlobalScope($scope, $identifier)
+    {
+        $this->scopes[$identifier] = $scope;
```

Comment:
There is an issue here with overriding earlier anonymous scopes. Also, `$identifier` is non-optional so your example won't work. I would also prefer having `$identifier` as first parameter as it reads nicer (you can still make it optional though)